Add a prototype release creation step

smarterclayton opened the pull request:
The step reuses the pod test logic (factored out) to run a cli command
to craft a new release from the latest stable stream in a PR. The release
payload is published to "release:latest" in the image stream. The step
is always run as long as there is a release tag specification. The step
does nothing if no image "cluster-version-operator" is present in the stable
image stream.

Templates can reference this via `RELEASE_IMAGE_LATEST`.  It is currently an optional target and only a few jobs will directly target it.

Design doc: https://docs.google.com/document/d/1CGZVEyuloZ9oD4NUArW6dnEpi0WFc6BP2tqPSwFZTCY/edit#heading=h.1zgrwxmpgxbr

stevekuznetsov commented:
Is there a declarative design doc for the format of the release?

stevekuznetsov commented on pkg/steps/release/create_release.go:
```diff
@@ -0,0 +1,139 @@
+	"github.com/openshift/ci-operator/pkg/steps"
+)
+
+// assembleReleaseStep will tag a full release suite
```
bad doc

stevekuznetsov commented on pkg/steps/release/create_release.go:
```diff
@@ -0,0 +1,139 @@
+		if !errors.IsAlreadyExists(err) {
+			return err
+		}
+		release, err = s.imageClient.ImageStreams(s.jobSpec.Namespace).Get("release", meta.GetOptions{})
```
failed create --> eat error --> try a get

why? this "works" when the create error is a permissions error and the target is already exists?

smarterclayton commented:
This is still opt in - once this merges I'm going to have the origin GCP jobs depend on this (which will ensure it is build).  Ideally, any repo that is an operator (builds an image with label "io.openshift.release.operator") would have this task run on it, but that can be a later step.

smarterclayton commented:
Updated as per Steves comments offline, fixed a bug where we were tracking the wrong images (we were using the release tag config, not the stable image stream).

smarterclayton commented:
Got verbal from Steve, merging so we can start testing this out in the jobs.
